.github/workflows/style.yml

```diff
@@ -1,7 +1,22 @@
 # check spelling, codestyle
 name: Style Check
 
-on: [push, workflow_dispatch]
+on:
+  pull_request:
+     branches-ignore:
+       - '*no-ci*'
+       - 'dependabot/*'
+  push:
+    tags:
+      - "*"
+    branches:
+      - master
+      - "release*"
+      - 'dependabot/*'
+
+concurrency:
```

Comment:
@PProfizi What's that for ?

Reply:
@anslpa Not sure which line you mean but:
The dependabot/* is to not run style check jobs on dependabot PRs (not needed)

The concurrency is to cancel previously running workflows for a PR when a new commit is made. This is to prevent clusters of jobs running for older commits than the latest. It is a choice, we can discuss tweaking that.